### PR TITLE
TSG/Services support

### PR DESCRIPTION
**Pull request by cheapRoc**
Ref: #127 

This is the continuation of the alpha TSG support for integrating with Terraform. The integration can be found in [my personal branch](https://github.com/cheapRoc/terraform-provider-triton/tree/tsg-alpha).

### CODE REVIEW

File: compute/instances.go [hunk 1]
```diff
@@ -1125,9 +1125,9 @@ var reservedInstanceCNSTags = map[string]struct{}{
 	CNSTagServices:   {},
 }
 
-// tagsExtractMeta() extracts all of the misc parameters from Tags and returns a
+// TagsExtractMeta() extracts all of the misc parameters from Tags and returns a
```
Comment (sean-) [inline compute/instances.go]: comment on exported function TagsExtractMeta should be of the form "TagsExtractMeta ..."

File: client/client.go [hunk 2]
```diff
@@ -31,8 +32,18 @@ var (
...
+	InvalidServicesURL = "invalid format of Triton Service Groups URL"
+	InvalidDCInURL     = "invalid data center in URL"
+
+	knownDCFormats = []string{
+		`https?://(.*).api.joyent.com`,
+		`https?://(.*).api.joyentcloud.com`,
+		`https?://(.*).api.samsungcloud.io`,
+	}
+
+	tsgFormatURL = "https://tsg.%s.svc.joyent.zone"
```
Comment (cheapRoc) [inline client/client.go]: We need to switch to `samsungcloud.zone` when the API is `samsungcloud.io`.

File: client/client.go [hunk 3]
```diff
@@ -455,11 +501,74 @@ func (c *Client) ExecuteRequestNoEncode(ctx context.Context, inputs RequestNoEnc
...
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "unable to execute HTTP request")
+	}
+
+	if resp.StatusCode >= http.StatusOK &&
+		resp.StatusCode < http.StatusMultipleChoices {
+		return resp.Body, nil
+	}
+
+	return nil, fmt.Errorf("could not process backend TSG request")
```
Comment (cheapRoc) [inline client/client.go]: Add proper request error handling here.

**Comment by cheapRoc** [comment]
Removed `HealthCheckInterval` and added unit tests for Service Group APIs.

**Comment by cheapRoc** [comment]
Removed `InstanceNamePrefix` and added unit tests for Instance Template APIs.

**Comment by cheapRoc** [comment]
Included timestamps merged into TSG and tested with templates example.

**Comment by cheapRoc** [comment]
Taking off the WIP but we still want to integrate with Terraform before we wrap this up, just for final sanity checks. 

Otherwise, I feel like CR can begin.

**Comment by cheapRoc** [comment]
Ok this works in the latest alpha TF provider support so I'm ready to get this merged.

**Comment by cheapRoc** [comment]
Ok that should satisfy the linter.

**Comment by stack72** [comment]
This LGTM! This has been largely reviewed as we go and has been tested against Terraform provider as well to make sure that it works as expected

**Pull request merged**